packages/client/src/OsdkObjectFrom.ts

```diff
@@ -123,7 +123,8 @@ export type Osdk<
   }
   // We are hiding the $rid field if it wasnt requested as we want to discourage its use
   & (IsNever<P> extends true ? {}
-    : P extends "$rid" ? { $rid: string }
+    : string extends P ? {}
```

Comment:
This is not as permissive, so passing in `any` will not give you access to the `rid` property unless you explicitly ask for it. This matches up with the behavior with our existing type tests which is why I kept it like this for now.